main/lsp/QueryCollector.cc

```diff
@@ -0,0 +1,57 @@
+#include "main/lsp/QueryCollector.h"
+
+using namespace std;
+namespace sorbet::realmain::lsp {
+namespace {
+// In the case of location ties, determines which query response takes priority in the vector constructed by
+// flushErrors. Larger values means greater specificity.
+u2 getQueryResponseTypeSpecificity(const core::lsp::QueryResponse &q) {
+    if (q.isEdit()) {
+        // Only reported for autocomplete, and should take precedence over anything else reported
+        return 8;
+    } else if (q.isDefinition()) {
+        return 7;
+    } else if (q.isSend()) {
+        return 6;
+    } else if (q.isField()) {
+        return 5;
+    } else if (q.isIdent()) {
+        return 4;
+    } else if (q.isConstant()) {
+        return 3;
+    } else if (q.isLiteral()) {
+        return 2;
+    } else {
+        return 1;
+    }
+}
+} // namespace
+void QueryCollector::flushErrors(spdlog::logger &logger, const core::GlobalState &gs, core::FileRef file,
+                                 vector<unique_ptr<core::ErrorQueueMessage>> errors) {
+    for (auto &error : errors) {
+        if (error->kind == core::ErrorQueueMessage::Kind::QueryResponse) {
+            queryResponses.emplace_back(move(error->queryResponse));
+        }
+    }
+
+    stable_sort(queryResponses.begin(), queryResponses.end(), [](auto &left, auto &right) -> bool {
```

Comment:
This function is now going to sort query responses every time they are flushed, which could be expensive for "Find All References" requests spanning many files.

Can we move this into a function called `drainQueryResponses()` that returns a sorted vector? Then you can make `queryResponses` private.